app/config.js

```diff
@@ -233,3 +235,174 @@ export function newWalletConfigCreation(testnet, walletPath) {
     ini.stringify(dcrwConf)
   );
 }
+
+export function checkNoLegacyWalletConfig(testnet, walletPath, noLegacyRpc) {
+  let fileContents;
+  try {
+    fileContents = ini.parse(
+      fs.readFileSync(dcrwalletConf(getWalletPath(testnet, walletPath)), "utf8")
+    );
+    fileContents = Object.fromEntries(
+      Object.entries(fileContents).map(([key, value]) => {
+        if (key == "Application Options") {
+          if (value.nolegacyrpc && noLegacyRpc) {
+            delete value.nolegacyrpc;
+          } else if (!noLegacyRpc) {
+            value.nolegacyrpc = "1";
+          }
+        }
+        return [key, value];
+      })
+    );
+    fs.writeFileSync(
+      dcrwalletConf(getWalletPath(testnet, walletPath)),
+      ini.stringify(fileContents)
+    );
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+export const getDefaultBitcoinConfig = () =>
+  new Promise((resolve, reject) => {
+    try {
+      const config = ini.parse(
+        fs.readFileSync(
+          path.join(getDefaultBitcoinDirectory(), "bitcoin.conf"),
+          "utf8"
+        )
+      );
+      resolve(config);
+    } catch (e) {
+      reject(e);
+    }
+  });
```

Comment:
Could be simplified by an async call (async functions implicitly return a promise - try `(async () => "he")();` in a modern browser and you'll get a promise as return value).

Also `getDefaultBitcoinConfig` isn't exactly accurate, since this returns the _current_ config as opposed to the _default_ config (I guess `Default` there refers to being from the default dir?). Anyway, the name is a minor issue, it's just that in the next function I see the same code being rewritten.

```suggestion
export const getDefaultBitcoinConfig = async () => {
  const btcConfPath = path.join(getDefaultBitcoinDirectory(), "bitcoin.conf");
  return ini.parse(fs.readFileSync(btcConfPath, "utf8");
}
```